### PR TITLE
Fixed several minor issues.

### PR DESCRIPTION
**Pull request by desilinguist**
- [RSMR-280] Add MLA & BibTex citations to README.
- [RSMR-287] Fix minor confusion matrix bug.
- [RSMR-267] Make language in data description notebook more general.
- [RSMR-132] Add test that excludes flags AND zero scores.
- [RSMR-277] Wrap long axis labels in subgroup plots.
- [RSMR-274] Use `FeatureSet.from_data_frame()` instead of `create_featureset_from_dataframe()`.
- [RSMR-286] Increase timeout to 60 minutes.
- Add `comparison_id` as required field for `rsmcompare`.
- Also add length check for all ID fields that are used as part of filenames (i.e, `rsmtool, `rsmeval`, and `rsmcompare`). This is now done in a new function called `check_id_fields()`.
- Update data all `rsmcompare` tests to have the `comparison_id` fields.
- Update documentation with `comparison_id` and length limits.
- Update `rsmcompare` example and tutorial.
- Update `test_input.py` to remove tests for `rsmpredict` and add new tests for ID lengths.
- Fix coveralls badge in README.

### CODE REVIEW

File: rsmtool/notebooks/features_by_group.ipynb [hunk 1]
```diff
@@ -47,6 +47,9 @@
     "    else:\n",
     "        box_names = ['All data'] + group_levels\n",
     "\n",
+    "    # wrap the names for each of the bars if they are longer then 20 characters\n",
```
Comment (aloukina) [inline rsmtool/notebooks/features_by_group.ipynb]: nitpick: bars ->boxes

File: rsmtool/report.py [hunk 2]
```diff
@@ -566,16 +574,15 @@ def create_comparison_report(experiment_id_old, description_old,
...
     # we define separate groups to allow future flexibility in defining
     # what groups are used for descriptives and evaluations
     os.environ['GROUPS_FOR_DESCRIPTIVES'] = '%%'.join(subgroups)
     os.environ['GROUPS_FOR_EVALUATIONS'] = '%%'.join(subgroups)
 
     # create the output directory
     os.makedirs(output_dir, exist_ok=True)
-    report_name = '{}_vs_{}.report'.format(experiment_id_old,
-                                           experiment_id_new)
+    report_name = '{}.report'.format(comparison_id)
```
Comment (aloukina) [inline rsmtool/report.py]: Elsewhere we do *_report.html
Do we want to keep the . or change it?

File: tests/test_input.py [hunk 3]
```diff
@@ -114,37 +114,68 @@ def test_validate_experiment_id_2():
...
+    data = {'experiment_id': 'this_is_a_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_long_id',
+            'train_file': 'data/rsmtool_smTrain.csv',
+            'test_file': 'data/rsmtool_smEval.csv',
+            'model': 'LinearRegression'}
+    validate_and_populate_json_fields(data)
+
+
+@raises(ValueError)
+def test_validate_experiment_id_6():
+    data = {'experiment_id': 'this is a really really really really really really really really really really really really really really really really really really really really really really really really really really really long id',
```
Comment (aloukina) [inline tests/test_input.py]: This one will fail because of the spaces, not the length - is that the idea?

**Comment by aloukina** [comment]
Other than the minor things above this looks good.

File: rsmtool/report.py (hunk 2 above)
Comment (desilinguist) [inline rsmtool/report.py]: Good idea, I will change it to `_report` to be consistent.

File: tests/test_input.py (hunk 3 above)
Comment (desilinguist) [inline tests/test_input.py]: It's basically that even though it has spaces, it will fail because of length first because of how the exception raising is ordered in `check_id_fields()`.
